include/zephyr/net/ieee802154_pkt.h

```diff
@@ -82,16 +95,89 @@ static inline void net_pkt_set_ieee802154_lqi(struct net_pkt *pkt, uint8_t lqi)
 	net_pkt_cb_ieee802154(pkt)->lqi = lqi;
 }
 
+/**
+ * @brief Get the unsigned RSSI value as defined in section 6.16.2.8,
+ *        Received Signal Strength Indicator (RSSI)
+ *
+ * @param pkt Pointer to the packet.
+ *
+ * @returns RSSI represented as unsigned byte value, ranging from
+ *          0 (–174 dBm) to 254 (80 dBm). 255 indicates that an
+ *          RSSI value is not available for this packet. Will return
+ *          zero for packets on the TX path.
+ */
 static inline uint8_t net_pkt_ieee802154_rssi(struct net_pkt *pkt)
 {
 	return net_pkt_cb_ieee802154(pkt)->rssi;
 }
 
+/**
+ * @brief Set the unsigned RSSI value as defined in section 6.16.2.8,
+ *        Received Signal Strength Indicator (RSSI).
+ *
+ * @param pkt Pointer to the packet that was received with the given
+ *            RSSI.
+ * @param rssi RSSI represented as unsigned byte value, ranging from
+ *             0 (–174 dBm) to 254 (80 dBm). 255 indicates that an
+ *             RSSI value is not available for this packet.
+ */
 static inline void net_pkt_set_ieee802154_rssi(struct net_pkt *pkt, uint8_t rssi)
 {
 	net_pkt_cb_ieee802154(pkt)->rssi = rssi;
 }
 
+/**
+ * @brief Get a signed RSSI value measured in dBm.
+ *
+ * @param pkt Pointer to the packet.
+ *
+ * @returns RSSI represented in dBm. Returns the special value
+ *          INT16_MIN if an RSSI value is not available for this
+ *          packet. Packets on the TX path will always show
+ *          -174 dBm (which corresponds to an internal value of
+ *          unsigned zero).
+ */
+static inline int16_t net_pkt_ieee802154_rssi_dbm(struct net_pkt *pkt)
+{
+	int16_t rssi = net_pkt_cb_ieee802154(pkt)->rssi;
+	return rssi == UINT8_MAX ? INT16_MIN : rssi - 174;
+}
+
+/**
+ * @brief Set the RSSI value as a signed integer measured in dBm.
+ *
+ * @param pkt Pointer to the packet that was received with the given
+ *            RSSI.
+ * @param rssi represented in dBm. Set to the special value
+ *             INT16_MIN if an RSSI value is not available for
+ *             this packet. Values above 80 dBm will be mapped to
+ *             80 dBm, values below -174 dBm will be mapped to
+ *             -174 dBm.
+ */
+static inline void net_pkt_set_ieee802154_rssi_dbm(struct net_pkt *pkt, int16_t rssi)
+{
+	if (likely(rssi >= -174 && rssi <= 80)) {
```

Comment:
Can you please `#define` physical layer constants for `-174` and `80` and use those here?

Reply:
Fixed.